model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/role/entity/JpaRoleCompositeEntity.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.storage.jpa.role.entity;
+
+import org.hibernate.annotations.Immutable;
+
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+/**
+ * This is a child table of {@link JpaRoleEntity} that is managed via named queries to avoid loading all its contents
+ * via a {@link javax.persistence.OneToMany} relation.
+ */
+@Entity
+@Table(name = "kc_role_composite", uniqueConstraints = {@UniqueConstraint(columnNames = {"role_id", "child_role_id"})})
+@Immutable
+@NamedQueries({
+        @NamedQuery(name="getChildrenRoles",
+                query="select rolecomposite.key from JpaRoleCompositeEntity rolecomposite where rolecomposite.key.roleId = :roleId"),
+        @NamedQuery(name="deleteChildFromCompositeRole",
+                query="delete from JpaRoleCompositeEntity rolecomposite where rolecomposite.key.roleId = :roleId and rolecomposite.key.childRoleId = :childRoleId"),
+        @NamedQuery(name="deleteChildrenFromCompositeRole",
+                query="delete from JpaRoleCompositeEntity rolecomposite where rolecomposite.key.roleId = :roleId"),
```

Comment:
I assume we want to use named queries here to be able achieve something what cannot be done via criteria query, correct?

Reply:
I picked named query as I find them more readable. I use the criteria queries only when I need to do something more dynamic than placeholders. 

Would you suggest to rewrite it using a criteria query?

Reply:
> Would you suggest to rewrite it using a criteria query?

no, I don't mind to have named queries, I was just wondering why it's used.